src/config/__tests__/settings.spec.ts

```diff
@@ -0,0 +1,228 @@
+  });
+
+  test('one option with embedded equal sign should change only one value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-version=1.23.6']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.6');
+    newPrefs.kubernetes.version = prefs.kubernetes.version;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('one option over two args should change only one value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-version', '1.23.7']);
+
+    expect(newPrefs.kubernetes.version).toBe('1.23.7');
+    newPrefs.kubernetes.version = prefs.kubernetes.version;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('boolean option to true should change only that value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-options-flannel=true']);
+
+    expect(newPrefs.kubernetes.options.flannel).toBeTruthy();
+    newPrefs.kubernetes.options.flannel = false;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('boolean option set to implicit true should change only that value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-suppressSudo']);
+
+    expect(newPrefs.kubernetes.suppressSudo).toBeTruthy();
+    newPrefs.kubernetes.suppressSudo = false;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('boolean option set to false should change only that value', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, ['--kubernetes-options-traefik=false']);
+
+    expect(newPrefs.kubernetes.options.traefik).toBeFalsy();
+    newPrefs.kubernetes.options.traefik = true;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('should change several values (and no others)', () => {
+    const newPrefs = settings.updateFromCommandLine(prefs, [
+      '--kubernetes-options-traefik=false',
+      '--kubernetes-suppressSudo',
+      '--portForwarding-includeKubernetesServices=true',
+      '--kubernetes-containerEngine=containerd',
+      '--kubernetes-port', '6444'
+    ]);
+
+    expect(newPrefs.kubernetes.options.traefik).toBeFalsy();
+    expect(newPrefs.kubernetes.suppressSudo).toBeTruthy();
+    expect(newPrefs.portForwarding.includeKubernetesServices).toBeTruthy();
+    expect(newPrefs.kubernetes.containerEngine).toBe('containerd');
+    expect(newPrefs.kubernetes.port).toBe(6444);
+
+    newPrefs.kubernetes.options.traefik = true;
+    newPrefs.kubernetes.suppressSudo = false;
+    newPrefs.portForwarding.includeKubernetesServices = false;
+    newPrefs.kubernetes.containerEngine = settings.ContainerEngine.MOBY;
+    newPrefs.kubernetes.port = 6443;
+    expect(newPrefs).toMatchObject(prefs);
+  });
+
+  test('should complain about a non-option', () => {
+    const arg = 'doesnt-start-with-dash-dash=some-value';
+
+    expect(() => {
+      settings.updateFromCommandLine(prefs, [arg]);
+    }).toThrowError(new RegExp(`Unexpected argument '${ arg }'`));
```

Comment:
We just need a substring match here; don't use a RegExp.
```suggestion
    }).toThrow(`Unexpected argument '${ arg }'`);
```

This also applies to all other error matches.

Reply:
yup